test/fuzzers/spvtools_as_fuzzer.cpp

```diff
@@ -18,18 +18,26 @@
 
 #include "source/spirv_target_env.h"
 #include "spirv-tools/libspirv.hpp"
+#include "test/fuzzers/random_generator.h"
 
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
-  if (size < sizeof(spv_target_env) + 1) return 0;
-
-  const spv_context context =
-      spvContextCreate(*reinterpret_cast<const spv_target_env*>(data));
-  if (context == nullptr) return 0;
+  if (size < 1) {
+    return 0;
+  }
```

Comment:
Should the 0 bytes case really be excluded? A target environment could be chosen totally at random in that case. Similar for the other fuzzer targets.

Reply:
Picking an arbitrary random number would eliminate reproducibility of the test case, since a different target env could be picked each time the fuzzer is run with the same input.

The 0 sized case feels like an edge case that is better for something like unit or other functional testing, since it is feels pretty tractable to exhaustively test all of different classes of cases, i.e. {nullptr, 0}, {valid pointer, 0}, {invalid pointer, 0}.

Reply:
I see what you mean about randomization.

I guess I'd favour including the 0-sized case, and using a specific target env in that scenario. I agree its an edge case, but it's the sort of edge case that people forget about with other testing methods, so wouldn't hurt to cover it here.

But I'm fine either way.

Reply:
Changing this to picking universal 1.0 in the 0 case.